internals/utils/helper.go

```diff
@@ -78,7 +81,7 @@ func AppendUpperResourcePoolAnnotation(sns *ObjectContext, parentSns *ObjectCont
 		}
 	}
 	upperRp, exists := parentSns.Object.GetAnnotations()[danav1.UpperRp]
-	if exists {
+	if exists && !isUpperResourcePool {
```

Comment:
Why is this needed?

Reply:
It makes sure that there won't be infinity api calls on the new upper resource pool